### PR TITLE
refactor: remove all lint warnings 

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: examples/examples.test.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable jest/expect-expect */
```
Comment (domoritz) [inline examples/examples.test.ts:1]: Can we silence the warning where is happens?

File: src/compile/compile.ts [hunk 2]
```diff
@@ -196,7 +196,7 @@ function assembleTopLevelModel(
   model: Model,
   topLevelProperties: TopLevelProperties & LayoutSizeMixins,
   datasets: Datasets = {},
-  usermeta: object
+  usermeta: Record<string, unknown>
```
Comment (domoritz) [inline src/compile/compile.ts:199]: Use dict instead. Also check against embed.

File: src/compile/legend/parse.ts [hunk 3]
```diff
@@ -61,7 +61,7 @@ function getLegendDefWithScale(model: UnitModel, channel: NonPositionScaleChanne
   return {[channel]: scale};
 }
 
-function isExplicit<T extends string | number | object | boolean>(
```
Comment (domoritz) [inline src/compile/legend/parse.ts:64]: Use dict instead.

File: src/compile/split.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import {deepEqual, duplicate, getFirstDefined, keys} from '../util';
  * This is important for scale/axis/legend merging as
  * we want to prioritize properties that users explicitly specified.
  */
-export class Split<T extends object> {
+export class Split<T extends Record<string, unknown>> {
```
Comment (domoritz) [inline src/compile/split.ts:10]: Use dict instead.

File: src/data.ts [hunk 5]
```diff
@@ -83,7 +83,13 @@ export type DataSource = UrlData | InlineData | NamedData;
 
 export type Data = DataSource | Generator;
 
-export type InlineDataset = number[] | string[] | boolean[] | object[] | string | object;
+export type InlineDataset =
+  | number[]
+  | string[]
+  | boolean[]
+  | Record<string, unknown>[]
```
Comment (domoritz) [inline src/data.ts:90]: Use dict

File: src/data.ts [hunk 6]
```diff
@@ -83,7 +84,7 @@ export type DataSource = UrlData | InlineData | NamedData;
 
 export type Data = DataSource | Generator;
 
-export type InlineDataset = number[] | string[] | boolean[] | object[] | string | object;
+export type InlineDataset = number[] | string[] | boolean[] | Dict<any>[] | string | Dict<any>;
```
Comment (domoritz) [inline src/data.ts:87]: Please revert this. See https://github.com/vega/vega-lite/pull/6588

File: src/compile/axis/parse.ts [hunk 7]
```diff
@@ -167,7 +167,7 @@ function mergeAxisComponent(merged: AxisComponent, child: AxisComponent): AxisCo
   return merged;
 }
 
-function isExplicit<T extends string | number | boolean | object>(
+function isExplicit<T extends string | number | boolean | Dict<any>>(
```
Comment (domoritz) [inline src/compile/axis/parse.ts:170]:
```suggestion
function isExplicit<T extends string | number | boolean | Dict<unknown>>(
```

We should be using unknown, not any here. That was what the lint warning was about, right?

File: src/util.ts [hunk 8]
```diff
@@ -209,6 +209,7 @@ export function fieldIntersection(a: ReadonlySet<string>, b: ReadonlySet<string>
   return hasIntersection(prefixGenerator(a), prefixGenerator(b));
 }
 
+// eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (kanitw) [reply src/util.ts:212]: Meh, it's just one line, not all lines.

[the remark this answers is not in the document]
Comment (domoritz) [reply src/util.ts:212]: Doesn't vscode automatically suggest the comment with the specific rule?

[the remark this answers is not in the document]
Comment (kanitw) [reply src/util.ts:212]: no

File: test-runtime/translate.test.ts [hunk 9]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable jest/expect-expect */
```
[the remark this answers is not in the document]
Comment (kanitw) [reply test-runtime/translate.test.ts:1]: All test in this file are problematic, so better put it here.

File: test-runtime/zoom.test.ts [hunk 10]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable jest/expect-expect */
```
[the remark this answers is not in the document]
Comment (kanitw) [reply test-runtime/zoom.test.ts:1]: Same as above lol

File: src/util.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (domoritz) [reply src/util.ts]: I see this quick fix action

Screen Shot 2020-06-21 at 21 37 54

File: src/compile/compile.ts [hunk 11]
```diff
@@ -196,7 +196,7 @@ function assembleTopLevelModel(
   model: Model,
   topLevelProperties: TopLevelProperties & LayoutSizeMixins,
   datasets: Datasets = {},
-  usermeta: object
+  usermeta: Dict<any>
```
Comment (domoritz) [inline src/compile/compile.ts:199]:
```suggestion
  usermeta: Dict<unknown>
```

Would this still work or should this also be `object`?

File: src/compile/legend/parse.ts [hunk 12]
```diff
@@ -61,7 +61,7 @@ function getLegendDefWithScale(model: UnitModel, channel: NonPositionScaleChanne
...
+function isExplicit<T extends string | number | Dict<any> | boolean>(
```
Comment (domoritz) [inline src/compile/legend/parse.ts:64]:
```suggestion
function isExplicit<T extends string | number | Dict<unknown> | boolean>(
```

File: src/compile/split.ts [hunk 13]
```diff
@@ -1,13 +1,13 @@
...
+import {deepEqual, Dict, duplicate, getFirstDefined, keys} from '../util';
 
 /**
  * Generic class for storing properties that are explicitly specified
  * and implicitly determined by the compiler.
  * This is important for scale/axis/legend merging as
  * we want to prioritize properties that users explicitly specified.
  */
-export class Split<T extends object> {
+export class Split<T extends Dict<any>> {
```
Comment (domoritz) [inline src/compile/split.ts:10]:
```suggestion
export class Split<T extends Dict<unknown>> {
```

File: src/data.ts [hunk 14]
```diff
@@ -196,7 +198,7 @@ export interface SphereGenerator extends GeneratorBase {
   /**
    * Generate sphere GeoJSON data for the full globe.
    */
-  sphere: true | {};
+  sphere: true | Dict<any>;
```
Comment (domoritz) [inline src/data.ts:201]:
```suggestion
  sphere: true | Dict<unknown>;
```

Or should this be `object` so we allow interface types?

File: src/util.ts [hunk 15]
```diff
@@ -15,7 +15,7 @@ export const duplicate = clone_;
  * pick(object, ['a', 'c']);
  * // → {'a': 1, 'c': 3}
  */
-export function pick<T extends object, K extends keyof T>(obj: T, props: readonly K[]): Pick<T, K> {
+export function pick<T extends Dict<any>, K extends keyof T>(obj: T, props: readonly K[]): Pick<T, K> {
```
Comment (domoritz) [inline src/util.ts:18]:
```suggestion
export function pick<T extends Dict<unknown>, K extends keyof T>(obj: T, props: readonly K[]): Pick<T, K> {
```

File: src/util.ts [hunk 16]
```diff
@@ -29,7 +29,7 @@ export function pick<T extends object, K extends keyof T>(obj: T, props: readonl
  * The opposite of _.pick; this method creates an object composed of the own
  * and inherited enumerable string keyed properties of object that are not omitted.
  */
-export function omit<T extends object, K extends keyof T>(obj: T, props: readonly K[]): Omit<T, K> {
+export function omit<T extends Dict<any>, K extends keyof T>(obj: T, props: readonly K[]): Omit<T, K> {
```
Comment (domoritz) [inline src/util.ts:32]:
```suggestion
export function omit<T extends Dict<unknown>, K extends keyof T>(obj: T, props: readonly K[]): Omit<T, K> {
```

**Pull request merged**